### ISSUE TITLE
Instrumented try...catch fails compilation step with Mythril

### ISSUE THREAD

**Issue by Thomas-Sciaroni** [ConsenSys/scribble#191]
Hello @cd1m0 !  We met last week at the formal reasoning workshop where you demoed scribble!

I am integrating it with our smart contracts.

I am instrumenting a contract that includes an OpenZeppelin ERC-1155.

After running scribble `scribble <file>.sol --output-mode flat --output <file>.flat.sol` I get a flattened file with syntax errors.

When _doSafeTransferAcceptanceCheck is instrumented:
```
    function _doSafeTransferAcceptanceCheck(address operator, address from, address to, uint256 id, uint256 amount, bytes memory data) private {
        if (to.isContract()) {
            try _callsite_5254(IERC1155Receiver(to), operator, from, id, amount, data) returns (bytes4 response) {
                if (response != IERC1155Receiver.onERC1155Received.selector) {
                    revert("ERC1155Receiver rejected");
                }
            } catch Error(string memory reason) {
                revert(reason);
            } catch {
                revert("non-ERC1155Receiver");
            }
        }
    }
```
the function _callsite_5254 is not an external call, though it contains one:
```
    function _callsite_5254(IERC1155Receiver receiver, address arg0, address arg1, uint256 arg2, uint256 arg3, bytes memory arg4) private returns (bytes4 ret0) {
        __scribble_check_state_invariants();
        __scribble_out_of_contract = true;
        (ret0) = receiver.onERC1155Received(arg0, arg1, arg2, arg3, arg4);
        __scribble_out_of_contract = false;
    }
 ```
 And the solidity compiler (invoked by mythril -- `myth analyze <file>.flat.sol -t 1` emits this error:
 ```
 mythril.interfaces.cli [ERROR]: Solc experienced a fatal error.

TypeError: Try can only be used with external function calls and contract creation calls.
    --> file.flat.sol:1325:17:
     |
1325 |             try _callsite_5254(IERC1155Receiver(to), operator, from, id, amount, data) returns (bytes4 response) {
     |            
```

I understand that this is a limitation of the solidity language.

for now I am removing the try...catch and using the code shown below:
```
    function _doSafeBatchTransferAcceptanceCheck(
        address operator,
        address from,
        address to,
        uint256[] memory ids,
        uint256[] memory amounts,
        bytes memory data
    ) private {
        if (to.isContract()) {
            bytes4 response = IERC1155Receiver(to).onERC1155BatchReceived(
                operator,
                from,
                ids,
                amounts,
                data
            );
            require(
                response == IERC1155Receiver.onERC1155BatchReceived.selector,
                "ERC1155Receiver rejected"
            );
        }
    }
```
I think this has the same behavior as the try...catch except for the message that is returned to the client when a call reverts.

Do you have any suggestions?

**Comment by cd1m0** [comment]
Thank you for the report! 
This looks like a big. Will try to reproduce shortly and get back to you.

**Comment by cd1m0** [comment]
Sorry for the delay. Candidate fix for your issue is under review here: https://github.com/ConsenSys/scribble/pull/196

**Comment by cd1m0** [comment]
Fixed in #196
